### PR TITLE
doc: readme: fix example conversion configs: all -> all_topics

### PR DESCRIPTION
**Pull request by ottojo**
this was changed in #1480, see issue #1660

### CODE REVIEW

**Review by fujitatomoya — APPROVED** [review]
good eye! lgtm.

@MichaelOrlov @Barry-Xu-2018 this actually breaks the user configuration as reported. i will go through the documentation again.

**Comment by Barry-Xu-2018** [comment]
The `RecordOptions` has already been changed quite a bit.  
Refer to the code below.

https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/rosbag2_transport/src/rosbag2_transport/record_options.cpp#L58-L103

The current README.md only mentions part of it. Is it necessary to point to the relevant code, so users know what can be configured ? @fujitatomoya @MichaelOrlov

https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/README.md?plain=1#L221-L240

**Comment by Barry-Xu-2018** [comment]
@ottojo 

Additionally.  ` exclude` should be changed to `exclude_regex`.  
Please refer to https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/rosbag2_transport/src/rosbag2_transport/record_options.cpp#L58-L103

**Comment by ottojo** [comment]
@Barry-Xu-2018 thanks for pointing me to the corresponding parts in the source. I updated the `exclude` key and added the other keys relevant for topic and service selection. I did not add all keys found in the source, i think some may already be excluded on purpose, because they are not relevant for bag conversion (such as for example `disable_keyboard_controls`, `compression_threads_priority` and `topic_polling_interval`)

**Comment by ottojo** [comment]
oh, i also just notice that it's not only the `RecordingOptions` but also `StorageOptions`, where the actual options are that i wanted to use prior to hitting this issue:
https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/rosbag2_storage/src/rosbag2_storage/storage_options.cpp#L44-L56

**Review by fujitatomoya — APPROVED** [review]
lgtm.

i do not think we need to introduce all options here, but all options need to be aligned with the implementation.

File: README.md [hunk 1]
```diff
@@ -247,7 +253,7 @@ $ ros2 bag convert -i bag1 -i bag2 -o out.yaml
 # out.yaml
 output_bags:
 - uri: merged_bag
-  all: true
+  all_topics: true
```
Comment (MichaelOrlov) [inline README.md:256]:
```suggestion
  all_topics: true
  all_services: true
```

File: README.md [hunk 2]
```diff
@@ -271,7 +277,7 @@ $ ros2 bag convert -i bag1 -o out.yaml
 # out.yaml
 output_bags:
 - uri: compressed
-  all: true
+  all_topics: true
```
Comment (MichaelOrlov) [inline README.md:281]:
```suggestion
  all_topics: true
  all_services: true
```

**Review by MichaelOrlov — COMMENTED** [review]
I think it will be valuable to add `all_services: true` for examples with bag compression and bag split. 
To avoid other potential questions in the future, like "Why services were lost during conversion?"

**Review by MichaelOrlov — APPROVED** [review]
LGTM. Thanks for the PR!

**Review by clalancette — APPROVED** [review]
Looks good to me as well.

Given that this is a change to documentation, we can merge this without CI, but @MichaelOrlov I'll wait for you to do that.  I think we should also backport this to the `jazzy` branch.

**Comment by MichaelOrlov** [comment]
Merging without CI run since changes only in the `readme.md` file.

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport jazzy

**Pull request merged**